### ISSUE TITLE
Files get uploaded twice

### ISSUE THREAD

**Issue by simarsenault** [eivindveg/HotSUploader#91]
The following happens on my machine (Windows 10):

I open HotS Replay Uploader then I play a few games (let say 3). Each replay is uploaded as soon as the game finishes. Then I close the JAR, reopen it and the 3 replays are uploaded again. However, if I close the JAR again and reopen it, the 3 replays won't be uploaded again.

If I have some time this week I'll fire up IntelliJ and try to find why this happens.

**Comment by zhedar** [comment]
Thanks for reporting, this is already a known issue we sinked several hours into.  
Our assumption is, that the h2 database is to blame, somehow the data isn't written to the store, if not enough bytes are in the buffer. However this may be completely wrong, as the data is saved correctly after the second run.  Changed the DBMS for 2.1 is a possible solution, but fixing this may be even better.
If you know something about that or just feel like trying around a bit, feel free to track this one down, any hints would be highly appreciated.

A note on that: Since we introduced a parser functionality in 2.0, the replays aren't really uploaded again. They are parsed, a single http request (for each replay file) is made and the file get's recognized as already uloaded.
